sklearn/metrics/pairwise.py

```diff
@@ -1276,14 +1271,45 @@ def pairwise_distances_chunked(X, Y=None, reduce_func=None,
                 is euclidean_distances):
             # zeroing diagonal, taking care of aliases of "euclidean",
             # i.e. "l2"
-            D_chunk.flat[sl.start::_num_samples(X) + 1] = 0
+            D_chunk.flat[sl.start::_num_samples(X) + 1] = 0.0
         if reduce_func is not None:
             chunk_size = D_chunk.shape[0]
             D_chunk = reduce_func(D_chunk, sl.start)
             _check_chunk_size(D_chunk, chunk_size)
         yield D_chunk
 
 
+def _set_diagonal_pairwise(distances, X, Y):
+    """Set the diagonal of unary pairwise distance to 0. This ensure that
+    the distances between themselves are always 0.0.
+
+    Parameters
+    ----------
+    X : array [n_samples_a, n_samples_a]
+        Array of pairwise distances between samples, or a feature array.
+
+    Y : array [n_samples_b, n_features]
+        Second feature array
+
+    distances : array [n_samples_a, n_samples_a] or [n_samples_a, n_samples_b]
+        A distance matrix D.
+    """
+    if Y is None or X is Y:
+        if not isinstance(X, list):
+            X_shape_0 = X.shape[0]
+        else:
+            X_shape_0 = 1
+        if not np.allclose(distances[np.diag_indices(X_shape_0)], 0.0):
```

Comment:
I assume that several tests fail because this allclose is too strict. Please report the maximum difference for each of the failing tests.

Reply:
Yes, that is the reason about the test fails. Sorry, I don't understand what do you say with report maximum difference. Sorry

Reply:
I meant that if you told me the max(abs(diagonal)) for each failing test, I would have a better understanding of how far off we were